### PR TITLE
[Form group] Better pt-control support

### LINKED ISSUE #1305: Switches in form groups have a margin-bottom
### Bug report

Latest, Chrome

#### Steps to reproduce

1. Put a switch inside a form group
2. Put some helper text below the switch
3. Boom, white space

#### Actual behavior

Lots of white space.

#### Expected behavior

Not so much white space.

### PR DESCRIPTION
**Pull request by llorca**
#### Changes proposed in this pull request:

- Proper `min-height` for `.pt-control` (16px normal, 20px large)
- Align `.pt-form-group` items to their baseline
- Tweak `.pt-control` vertical margin within `.pt-form-group`

#### Screenshot

Before:
screen shot 2017-07-11 at 4 15 18 pm

After:
screen shot 2017-07-11 at 4 15 26 pm

### CODE REVIEW

**Comment by cmslewis** [comment]
Verified in different browsers. Looks good!

**Comment by llorca** [comment]
Hehe, did you notice that, for inline form groups, the label text and input text (not switch) are off by 2px? I could not find any elegant fix

**Comment by cmslewis** [comment]
@llorca should we fix that before merging this? Figure we might as well throw 30 minutes at the problem now and hope for a resolution that obviates the need for future follow-up.

**Comment by llorca** [comment]
@cmslewis I would love to fix it and would be willing to do it before merging this, I just couldn't figure out a good solution (i.e. w/o nasty 2px margins here and there). If you have 10mins, mind taking a look? Happy to pair with you!

**Comment by llorca** [comment]
Thanks to @cmslewis, we no longer have alignment issues 🎉

File: packages/core/src/components/forms/_form-group.scss [hunk 1]
```diff
@@ -86,6 +119,12 @@ Styleguide pt-form-group
       margin: 0 $pt-grid-size 0 0;
       line-height: $pt-input-height;
     }
+
+    // not ideal but baseline doesn't align text perfectly
+    // so we have to align it to the label manually
+    .pt-control {
+      margin-top: $pt-grid-size * 1.2;
```
Comment (giladgray) [inline packages/core/src/components/forms/_form-group.scss]: ❌ absolutely not. try removing `.pt-large` from the example and it breaks...
image

gonna have to do this in a smarter way. perhaps make the height equal to `$input-height[-large]` (same as label)? will take some work to get this right...

File: packages/core/src/components/forms/_form-group.scss [hunk 2]
```diff
@@ -46,19 +60,38 @@ Markup:
...
+<div class="pt-form-group pt-inline pt-large pt-disabled">
+  <label class="pt-label" for="example-form-group-switch-b">
+    Label B
+    <span class="pt-text-muted">(required)</span>
+  </label>
+  <div class="pt-form-content">
+    <label class="pt-control pt-switch">
+      <input id="example-form-group-switch-b" type="checkbox" />
+      <span class="pt-control-indicator"></span>
+      Switch B
```
Comment (giladgray) [inline packages/core/src/components/forms/_form-group.scss]: use the correct next letter please. A B C D E

**Comment by cmslewis** [comment]
Ah yeah, @llorca, definitely add `margin-top: 1.2 * $pt-grid-size` only for `.pt-large`. The margin in the default case should be smaller (looking specifically at the `pt-switch`.

File: packages/core/src/components/forms/_form-group.scss (hunk 1 above)
Comment (llorca) [inline packages/core/src/components/forms/_form-group.scss]: That didn't work, the `.pt-control` markup doesn't vertically centers. I went back to `baseline`, it's literally the right tool for the job, I suspect the alignment problems are browser issues, specifically regarding text baseline _within_ an input. I can live with that if you guys can too, check out the new preview

**Pull request merged**